### ISSUE TITLE
Feature request: easier installation

### ISSUE THREAD

**Issue by jcbhmr** [Slackadays/Clipboard#32]
**TL;DR:** I would like to install this in one command, not like 10.

Right now the way to install this project is:

```sh
command -v cmake || sudo apt install cmake

temp_dir="$(mktemp -d)"
pushd "$temp_dir"

# https://github.com/Slackadays/Clipboard#clone-configure--compile
git clone https://github.com/slackadays/Clipboard
cd Clipboard
cmake .
cmake --build .
# https://github.com/Slackadays/Clipboard#install
sudo cmake --install .

popd
rm -rf "$temp_dir"
```

And that was after a full 5 minutes of head scratching before I saw the "Install" section as separate from the "Clone, configure, & compile" section and realized "OHHH I need to do the `--install` one too!"

Instead, I'd like to see one of these:

```sh
sudo apt install clipboard # Or whatever name
curl -sS https://webi.sh/clipboard | sh # webinstall.dev for more info
curl -sSL https://github.com/Slackadays/Clipboard/raw/main/install.sh | bash # Or a branded shortlink
```

And of course the Windows counterpart might be using `choco` or `Invoke-WebRequest | powershell` or whatever.

## Why this would be a good feature

Any good product makes it as easy as possible to onboard. You want to reduce the friction of your users getting started in order to make it "this is easy, so I might as well try it" or "its cheap to set up, so I'll try it!". The point is to **make it really really easy**. I think the current state is a bit challenging for someone who doesn't dive into the C++ ecosystem very much, but wants to use this as a CLI tool.

**TL;DR:** Make it more accessible!

<details>
  <summary>Suggestion for a readme section on "Installation"</summary>

![image](https://user-images.githubusercontent.com/61068799/210187612-16e4bbff-55a0-4d82-8468-f7642a347bb5.png)

But I understand if the design text or whatever gets changed. The point is that its _much_ easier for a user to copy-paste something like that than it is to think:
1. Do I have cmake installed?
2. How do I install cmake?
4. Is this worth it for a tool I just wanted to copy a `git log` output?
5. Ok now I need a temp folder to clone this into
6. What were the commands again to build it?
7. How do I get the binary onto my $PATH?

</details>

## Alternative solutions

Besides putting it all in an install.sh or install.ps1 script, you could also host that script through the https://webinstall.dev/ project. Another option is to use the system package manager. I have absolutely _no clue_ how to get involved with either of these; they're just ideas.

The simplest starting point, though, seems to be the self-hosted install.sh or install.ps1. You can even make it shorter with something like https://t.ly/

**Comment by jcbhmr** [comment]
Sidenote: is there a reason that this repo uses "Bug:" and "Feature request:" prefixes instead of tags? There's got to be a reason; I'm just curious.

**Comment by Slackadays** [comment]
I've thought a lot about making Clipboard a system package, but there there are a couple problems right now:
- each package system is a special snowflake with its own rules and conventions and requirements and format, so catering to each one will be a monumental task for me personally
- I read that most such packages aren't actually maintained by their authors but rather by package maintainers, and I've only seen two maintainers so far taking up the task (for Alpine and Arch Linux)
- integrating with native GUI clipboards is already a funky process with lots of special snowflake systems again, so that's another time vampire that is arguably more important than system packages right now

If I can get a package maintainer for Debian or Ubuntu, then that's half the battle done.

For the tagging situation, I just haven't bothered to set up tags yet because there have been so few issues so far. I noticed that a lot of projects of this popularity have issues and PRs piling up but I want to ensure that neve happens for Clipboard, so hopefully there'll be no more than 1-2 issues/PRs in the queue.

**Comment by Slackadays** [comment]
For the install script:
- Assume we have CMake and the right compiler
- Add a check for OpenBSD
- .sh for Linux/BSD, .ps1 or .bat for Windows?

Readme:
- Make the install script the main highlight of the QI section, while condensing the compile instructions to ignore platform specific additions

**Comment by Slackadays** [comment]
One-step install scripts for SH and Powershell have been added, I will have to make a special URL sometime in the future when I can set that up.

**Comment by jcbhmr** [comment]
Awesome! 👍 Really fast turnaround! 😲
